File: python/Unidad_2/ej6.py
# Realice operaciones aritméticas usando dos imágenes. Las imágenes deben 
# tener la misma dimensión para poder operar.

import matplotlib.pyplot as plt
from skimage.io import imread
import numpy as np
import tkinter as tk
from tkinter import filedialog

def plot_img(img, num_plot, title):
    '''
    Plot de imagen
    '''
    plt.subplot(num_plot)
    plt.imshow(img)
    plt.title(title)
    
def file_read():
    file_path = filedialog.askopenfilename()
    return file_path
    

try:
    # Busco imagen y obtengo su ruta
    root = tk.Tk()
    root.withdraw()
    img1 = imread(file_read()) # Female
    img2 = imread(file_read()) # Tree
    
    suma = img1 + img2
    resta = img1 - img2
    multip = (img1 * img2).astype(np.uint8)
    comb = (img1 * 0.3 + img2 * 0.7).astype(np.uint8)
    
    plot_img(img1, 231, 'Mujer')
    plot_img(img2, 232, 'Árbol')
    plot_img(suma, 233, 'Suma')
    plot_img(resta, 234, 'Resta')
    plot_img(multip, 235, 'Multiplicación')
    plot_img(comb, 236, 'Combinación')
    
    plt.show()
except:
    print('Cerraste la ventana!')

print('👋🏽')
root.destroy()

File: matlab/Unidad_2/ej6.m
% 6) Realice operaciones aritméticas usando dos imágenes. Las imágenes 
% deben tener la misma dimensión para poder operar. 

dirs = {}; % Creo una Cell para almacenar las dirs de las imagenes

for i=1:2
    [file, path] = uigetfile({'*.jpg';'*.JPG';'*.bmp';'*.tiff'});
    dir = strcat(path,file);
    info = imfinfo(dir);
    dirs{i} = im2double(imread(dir));
end

% Hay que utilizar imagenes del mismo tamaño
subplot(2,3,1);
imshow(dirs{1});
title('Imagen original 1');

subplot(2,3,2);
imshow(dirs{2});
title('Imagen original 2');

subplot(2,3,3);
imshow(dirs{1} + dirs{2});
title('Suma');

subplot(2,3,4);
imshow(dirs{1} - dirs{2});
title('Resta');

subplot(2,3,5);
imshow(dirs{1} .* dirs{2});
title('Producto');

subplot(2,3,6);
imshow(dirs{1} ./ dirs{2});
title('Division');